src/query/catalog/src/table_context.rs

```diff
@@ -111,5 +111,7 @@ pub trait TableContext: Send + Sync {
     // Get user manager api.
     fn get_user_manager(&self) -> Arc<UserApiProvider>;
     fn get_cluster(&self) -> Arc<Cluster>;
+    async fn get_table(&self, catalog: &str, database: &str, table: &str)
```

Comment:
Maybe we don't need this method here? Since we usually get table from `dyn Catalog`, which can be retrieved by `get_catalog`.

Reply:
> Maybe we don't need this method here? Since we usually get table from `dyn Catalog`, which can be retrieved by `get_catalog`.

`QueryContext` has some internal logic to keep the returning `Table` consistent throughout the query.

https://github.com/datafuselabs/databend/blob/f2a8f0da2e966376b954bf7e357ff0c4ce86704f/src/query/service/src/sessions/query_ctx_shared.rs#L190-L210

It's OK to remove it if we have other ways to make sure that.

Reply:
We do need some refactor for our `Catalog` and `TableContext` to make the behavior consistent.

Cc @dantengsky @zhang2014